default.py

```diff
@@ -24,9 +26,42 @@
     "serveropenload": "OpenLoad",
 }
 
+WATCHLIST_ITEMS = [
+    ("All", "all"),
+    ("Watching", "watching"),
+    ("Completed", "completed"),
+    ("On-Hold", "onhold"),
+    ("Dropped", "dropped"),
+    ("Plan To Watch", "planned")
+]
+
 _BROWSER = NineAnimeBrowser()
 control.setContent('tvshows');
 
+params = dict(urlparse.parse_qsl(sys.argv[2].replace('?', '')))
+print "params", params
+
+try: action = params['action']
+except: action = None
+
+try: anime_id = params['anime_id']
+except: anime_id = None
+
+try: folder = params['folder']
+except: folder = None
+
+if action == 'login':
```

Comment:
login class & handling should be part of the Browser module not lib.ui
can you refactor that plz?

UI is more kodi-generic stuff, not specific into a certain website..

Reply:
also, i've written `@route` directive long ago but i think you should be able to use it for the login instead of adding another parsing step